### PR TITLE
Duplicate response processing causes schema corruption

### PR DESCRIPTION
**Pull request by VitaliyKurokhtin**

### CODE REVIEW

**Comment by PerthCharern** [comment]
@VitaliyKurokhtin 

Any chance you have an example handy to illustrate what exactly is turning out incorrectly? 

I believe @darrelmiller added this change to allow reading `produces` if it comes after `responses`, but seems like it's causing some kind of regression. (#267)

**Comment by VitaliyKurokhtin** [comment]
I'll add a test, but in short the problem is that ProcessProduces fetches schema by context.GetFromTempStorage<OpenApiSchema>(TempStorageKeys.ResponseSchema) which is, if responses are already processed, the last schema. And that is the schema that gets assigned to all responses. I'm not sure how current implementation helps in supporting global produces, since produces is fetched from temp storage as well and If that information is available after operation and its responses are parsed, it must be available while they're being parsed as well. I'll talk to @darrelmiller.

**Comment by darrelmiller** [comment]
I tested this before and after the fix and was able to repro the problem and confirm the fix solves it.  Much appreciated @VitaliyKurokhtin !

File: test/Microsoft.OpenApi.Readers.Tests/Microsoft.OpenApi.Readers.Tests.csproj [hunk 1]
```diff
@@ -12,6 +12,10 @@
         <SignAssembly>true</SignAssembly>
         <AssemblyOriginatorKeyFile>..\..\src\Microsoft.OpenApi.snk</AssemblyOriginatorKeyFile>
     </PropertyGroup>
+    <ItemGroup>
```
Comment (PerthCharern) [inline test/Microsoft.OpenApi.Readers.Tests/Microsoft.OpenApi.Readers.Tests.csproj]: Could you remove this?

**Comment by PerthCharern** [comment]
Thanks @darrelmiller & @VitaliyKurokhtin 

@VitaliyKurokhtin Could you help resolve the merge conflict by removing all the <None Remove> tags and verify that there is no impact on your code and tests? Once you confirm that, I'll check in the code. Thank you for this fix :)

**Comment by VitaliyKurokhtin** [comment]
Sure, done!

**Comment by tobindh** [comment]
There is an issue with this fix if there are multiple produces. It clears the temp storage for the responseSchema in temp storage inside of the produces foreach, so if there are multiple content types (i.e. xml and json), only the first content type gets the schema assigned.  Shouldn't this clear the temp storage outside of the produces foreach?

**Comment by darrelmiller** [comment]
@tobindh Moving the clear to outside the Produces loop solves the problem with multiple produces, but it then breaks the issue with multiple responses.  Not sure why.  I will continue investigating.

**Comment by VitaliyKurokhtin** [comment]
@tobindh fixed by #398

**Pull request merged**